consensus/src/chained_bft/event_processor.rs

```diff
@@ -395,29 +403,16 @@ impl<T: Payload> EventProcessor<T> {
             Some((vote, vote_round)) if (*vote_round == round) => Some(vote.clone()),
             _ => {
                 // Try to generate a backup vote
-                let backup_vote_res = self.gen_backup_vote(round).await;
-                match &backup_vote_res {
-                    Ok(backup_vote_msg) => {
-                        self.last_vote_sent
-                            .replace((backup_vote_msg.clone(), round));
+                match self.gen_backup_vote(round).await {
+                    Ok(backup_vote_msg) => Some(backup_vote_msg),
```

Comment:
as the vote becomes mandatory in Timeout, we probably don't need the Some(..) here